### PR TITLE
[1.19.x] Re-implement RenderTooltipEvent.Color

### PR DESCRIPTION
**Pull request by XFactHD**
This PR re-implements the firing of `RenderTooltipEvent.Color`. This was previously disabled due to the tooltip background rendering being moved to `TooltipRenderUtil`. The existing test in `CustomTooltipTest` (modifies the tooltip of the Apple) was used to test this.

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/gui/screens/inventory/tooltip/TooltipRenderUtil.java.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ 
+    private static void m_262846_(TooltipRenderUtil.BlitPainter p_262974_, Matrix4f p_263030_, BufferBuilder p_263061_, int p_262982_, int p_262912_, int p_262946_, int p_263049_, int p_263118_, int p_263104_, int p_262979_) {
+@@ -49,8 +_,9 @@
+       p_262966_.m_262815_(p_263094_, p_262943_, p_262949_, p_263068_, p_262949_ + p_263111_, p_263068_ + 1, p_263095_, p_262918_, p_262918_);
+    }
+ 
+-   private static void m_262805_(TooltipRenderUtil.BlitPainter p_262933_, Matrix4f p_263035_, BufferBuilder p_263084_, int p_262945_, int p_262955_, int p_263017_, int p_262978_, int p_262950_, int p_262931_) {
+-      p_262933_.m_262815_(p_263035_, p_263084_, p_262945_, p_262955_, p_262945_ + p_263017_, p_262955_ + p_262978_, p_262950_, p_262931_, p_262931_);
++   // Forge: Allow specifying a gradient instead of a single color for the background
++   private static void renderRectangle(TooltipRenderUtil.BlitPainter p_262933_, Matrix4f p_263035_, BufferBuilder p_263084_, int p_262945_, int p_262955_, int p_263017_, int p_262978_, int p_262950_, int p_262931_, int colorTo) {
```
Comment (ChampionAsh5357) [inline patches/minecraft/net/minecraft/client/gui/screens/inventory/tooltip/TooltipRenderUtil.java.patch]: Don't remove methods, deprecate and redirect.

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]
Does `RenderTooltipEvent$Color` only specify vertical gradients?

**Comment by XFactHD** [comment]
`RenderTooltipEvent$Color` is indeed limited to vertical gradients due to `GuiComponent#fillGradient()` being limited to vertical gradients and no alternative method for horizontal or mixed gradients existing.

File: src/main/resources/forge.exc [hunk 2]
```diff
@@ -42,10 +44,10 @@ net/minecraft/data/tags/FluidTagsProvider.<init>(Lnet/minecraft/data/PackOutput;
...
+net/minecraft/data/tags/IntrinsicHolderTagsProvider.<init>(Lnet/minecraft/data/PackOutput;Lnet/minecraft/resources/ResourceKey;Ljava/util/concurrent/CompletableFuture;Ljava/util/function/Function;Ljava/lang/String;Lnet/minecraftforge/common/data/ExistingFileHelper;)V=|p_256164_,p_256155_,p_256488_,p_256168_,modId,existingFileHelper
 net/minecraft/data/tags/ItemTagsProvider.<init>(Lnet/minecraft/data/PackOutput;Ljava/util/concurrent/CompletableFuture;Ljava/util/concurrent/CompletableFuture;Ljava/lang/String;Lnet/minecraftforge/common/data/ExistingFileHelper;)V=|p_275343_,p_275729_,p_275322_,modId,existingFileHelper
+net/minecraft/data/tags/ItemTagsProvider.<init>(Lnet/minecraft/data/PackOutput;Ljava/util/concurrent/CompletableFuture;Ljava/util/concurrent/CompletableFuture;Ljava/util/concurrent/CompletableFuture;Ljava/lang/String;Lnet/minecraftforge/common/data/ExistingFileHelper;)V=|p_275204_,p_275194_,p_275207_,p_275634_,modId,existingFileHelper
```
Comment (ChampionAsh5357) [inline src/main/resources/forge.exc:50]: Nonrelevant changes to the PR, but other than that works fine and looks pretty much like the previous solution in a different location.

Comment (SizableShrimp) [reply src/main/resources/forge.exc:50]: I told him it was fine to include given it was automatically generated.

Comment (ChampionAsh5357) [reply src/main/resources/forge.exc:50]: Ah ok.

**Review by ChampionAsh5357 — APPROVED** [review]
Well, Shrimp addressed my comments, so it's good now.

In general though, this will probably need an overhaul again in 1.20 because of the new GuiGraphics system, so at least this will be fine for LTS users.

**Pull request merged**